tests/test_steerable_pyr.py

```diff
@@ -84,212 +86,170 @@ def check_parseval(im ,coeff, rtol=1e-4, atol=0):
+                                              downsample=eval(downsample), tight_frame=eval(tightframe))
+        pyr.to(DEVICE)
+        return pyr
+
+    # can't use one of the spyr fixtures here because we need to instantiate separately for each of these shapes
     @pytest.mark.parametrize("height", ['auto', 1, 3, 4, 5])
     @pytest.mark.parametrize("order", [1, 2, 3])
     @pytest.mark.parametrize('is_complex', [True, False])
     @pytest.mark.parametrize("im_shape", [None, (255, 255), (256, 128), (128, 256), (255, 256),
                                           (256, 255)])
-    def test_pyramid(self, height, order, is_complex, im_shape):
-        x = po.make_basic_stimuli()
+    def test_pyramid(self, basic_stim, height, order, is_complex, im_shape):
         if im_shape is not None:
-            x = x[..., :im_shape[0], :im_shape[1]]
-        spc = po.simul.Steerable_Pyramid_Freq(x.shape[-2:], height=height, order=order,
+            basic_stim = basic_stim[..., :im_shape[0], :im_shape[1]]
+        spc = po.simul.Steerable_Pyramid_Freq(basic_stim.shape[-2:], height=height, order=order,
                                               is_complex=is_complex)
-        spc(x)
+        spc(basic_stim)
+
+    @pytest.mark.parametrize('spyr', [f'{h}-{o}-{c}-{d}-True' for h, o, c, d in product(['auto', 1, 2, 3],
+                                                                                        [1, 2, 3],
+                                                                                        [True, False],
+                                                                                        [True, False])],
+                             indirect=True)
+    def test_tight_frame(self, img, spyr):
+        spyr.forward(img)
+        check_parseval(img, spyr.pyr_coeffs)
+
+    @pytest.mark.parametrize('spyr', [f'{h}-{o}-{c}-True-True' for h, o, c in product([3, 4, 5],
+                                                                                      [1, 2, 3],
+                                                                                      [True, False])],
+                             indirect=True)
+    def test_not_downsample(self, img, spyr):
+        spyr.forward(img)
+        # need to add 1 because our heights are 0-indexed (i.e., the lowest
+        # height has k[0]==0)
+        height = max([k[0] for k in spyr.pyr_coeffs.keys() if isinstance(k[0], int)]) + 1
+        # couldn't come up with a way to get this with fixtures, so we
+        # instantiate it each time.
+        spyr_not_downsample = po.simul.Steerable_Pyramid_Freq(img.shape[-2:], height, spyr.order,
+                                                              is_complex=spyr.is_complex,
+                                                              downsample=False, tight_frame=True)
+        spyr_not_downsample.to(DEVICE)
+        spyr_not_downsample.forward(img)
+        check_band_energies(spyr.pyr_coeffs, spyr_not_downsample.pyr_coeffs)
```

Comment:
this all looks good to me.  A more general point about these tests (not necessarily for your PR) I was thinking about is do we really need the tests that test the same shape but just transposed? For example, an image that is (128, 256) and (256, 128) will have the same scales because we set scales based on the min size. And I'm not sure that there is really anything else that would change in the pyramid. I'm not sure what the current runtime is for the pyr tests but if it is long, then these tests feel a bit redundant to me but it's not a big deal either way.

Reply:
The pyramid tests take 5 minutes, so I think the amount of time is fine. And I'd lean towards keeping them in because they should be the same, so it's worth checking to make sure -- it would be really bad if they weren't. I based this off the existing tests, so I don't know if there was a reason why we were explicitly testing both variants beforehand.

Reply:
Yeah this wasn't about your PR, was mostly about why we were doing that from the beginning. but if it's 5 mins then doesn't matter, but if we add more tests down the line and it ends up being cumbersome, maybe worth revisiting.